datafusion/sqllogictest/test_files/join.slt

```diff
@@ -998,11 +997,22 @@ CREATE TABLE t2 (v0 DOUBLE) AS VALUES (-1.663563947387);
 statement ok
 CREATE TABLE t3 (v0 DOUBLE) AS VALUES (0.05112015193508901);
 
+# Test issue: https://github.com/apache/datafusion/issues/11269
 query RR
 SELECT t3.v0, t2.v0 FROM t1,t2,t3 WHERE t3.v0 >= t1.v0;
 ----
 0.051120151935 -1.663563947387
 
+# Test issue: https://github.com/apache/datafusion/issues/11414
+query IRR
+SELECT * FROM t1 INNER JOIN t2 ON NULL RIGHT JOIN t3 ON TRUE;
+----
+NULL NULL 0.051120151935
+
+# ON expression must be boolean type
+query error DataFusion error: Error during planning: ON expression must be boolean, but got Utf8
+SELECT * FROM t1 INNER JOIN t2 ON 'TRUE'
```

Comment:
This query is available in PostgreSQL, but not available in Spark.
If we plan to support it, we can directly cast the ON expression to boolean without checking its current type.

Reply:
I think the error is clear and is a reasonable behavior. If it is important for someone's usecase, perhaps we can add support for it then

Reply:
I think the structure you have now implemented (in the analyzer) makes it pretty easy to support if desired